media/js/base/mozilla-fxa-link.js

```diff
@@ -34,6 +34,49 @@ if (typeof window.Mozilla === 'undefined') {
         return href;
     };
 
+    /**
+     * Intercept event handler for FxA links and forms, lets the browser drive the FxA Flow using
+     * the `showFirefoxAccounts` UITour API. Attaches several UTM parameters from the current page
+     * that will be forwarded to the browser and later on to FxA services.
+     * @param event {Event}
+     * @private
+     */
+    FxaLink.interceptFxANavigation = function(event) {
+        event.preventDefault();
+        var urlParams = new window._SearchParams().utmParams();
```

Comment:
We already have some logic for adding utm parameters automatically to Fxa links in [fxa-utm-referrer.js](https://github.com/mozilla/bedrock/blob/master/tests/unit/spec/base/fxa-utm-referral.js). Rather than reconstruct things again from the URL, might it be easier to get the existing query string directly from the link?

This way you would also pick up flow_id params that might also exist there, which seem to get missed with the current logic here? (unles I'm misreading something).

Reply:
> which seem to get missed with the current logic here?

The `fxa-link` doesn't have `flow_id`, only the form has the flow data.

Reply:
We add flow params to primary call-t-action links to FxA. This is handled by [`mozilla-fxa-product-button.js`](https://github.com/mozilla/bedrock/blob/master/media/js/base/mozilla-fxa-product-button.js), so something we still need to accomodate for here.